sfs/mono/source.py

```diff
@@ -79,6 +79,54 @@ def point_velocity(omega, x0, n0, grid, c=None):
     return util.XyzComponents([v * o / r for o in offset])
 
 
+def point_dipol(omega, x0, n0, grid, c=None):
+    """Point source with dipole characteristics.
+
+    Parameters
+    ----------
+    omega : float
+        Frequency of source.
+    x0 : (3,) array_like
+        Position of source.
+    n0 : (3,) array_like
+        Normal vector (direction) of dipole.
+    grid : triple of numpy.ndarray
+        The grid that is used for the sound field calculations.
+        See :func:`sfs.util.xyz_grid`.
+    c : float, optional
+        Speed of sound.
+
+    Returns
+    -------
+    numpy.ndarray
+        Sound pressure at positions given by `grid`.
+    ::
+
+         d                1   / iw       1    \   (x-x0) n0
+        ---- G(x-x0,w) = --- | ----- + ------- | ----------- e^(-i w/c |x-x0|)
+        d ns             4pi  \  c     |x-x0| /   |x-x0|^2
+
+    Examples
+    --------
+    .. plot::
+        :context: close-figs
+
+        p = sfs.mono.source.point_dipol(omega, x0, n0, grid)
+        sfs.plot.soundfield(p, grid)
+        plt.title("Dipole Point Source at {} m".format(x0))
+
+    """
+    k = util.wavenumber(omega, c)
+    x0 = util.asarray_1d(x0)
+    n0 = util.asarray_1d(n0)
+    grid = util.XyzComponents(grid)
+
+    r = np.linalg.norm(grid - x0)
+    scalar = np.inner((grid - x0), n0)
```

Comment:
`grid - x0` is repeated.
I called this `offset` in some places, but there are other places where it's called `dx`.
We should probably use only one of them.